### ISSUE TITLE
Links for typedef structs are missing

### ISSUE THREAD

**Issue by hroncok** [michaeljones/breathe#184]
I've run into a problem with links not being generated for types defined as follows:

```c
/** Normal vector of a facet, defined by 3D coordinates */
typedef struct {
  float x;
  float y;
  float z;
} stl_normal;
```

Now when I have another struct that has `stl_normal` typed members, the type name (`stl_normal`) does not appear as a link.

However, when I have let's say enum like this:

```c
/** Type of STL file */
typedef enum {binary, ascii, inmemory} stl_type;
```

All member variables of other structs typed as `stl_type` have the type as link:

```c
typedef struct {
  float x; // "float" is not a link, because float is a primitive type
  stl_type y; // "stl_type" is a link to it's description
  stl_normal z; // "stl_normal" is not a link, why?
} foo;
```

My Doxyfile is:

```
PROJECT_NAME           = ADMesh
STRIP_FROM_PATH        = ../src
STRIP_FROM_INC_PATH    = ../src
JAVADOC_AUTOBRIEF      = YES
QT_AUTOBRIEF           = YES
INPUT                  = ../src/admesh
GENERATE_LATEX         = NO
GENERATE_HTML          = NO
GENERATE_XML           = YES
XML_OUTPUT             = _doxyxml
OPTIMIZE_OUTPUT_FOR_C  = YES
TYPEDEF_HIDES_STRUCT   = YES # if I change this, it doesn't help
EXTRACT_ALL            = YES
QUIET                  = YES
```

When I set `GENERATE_HTML` to `yes` to see how normal HTML Doxygen output looks like, all links are there as expected.

Feel free to look here if my description is hard to understand: http://admesh.readthedocs.org/en/latest/c_api.html#stl_facet (but the content might change)

Based on the mailing list discussion, this is probably not my wrong-doing, but a bug.

**Comment by michaeljones** [comment]
Thank you for reporting this @hroncok. I can't look into it now but I will try to next week.

**Comment by vitaut** [comment]
This is fixed in Sphinx version 1.3.

**Comment by hroncok** [comment]
Thanks, indeed it is.
